### PR TITLE
Use pre-built artifact for tests

### PR DESCRIPTION
**Pull request by valerybugakov**
Creating a draft PR to share my changes based on [the discussion](https://sourcegraph.slack.com/archives/C01C3NCGD40/p1693809866106999) in Slack. Changes are focused on running tests and do not consider the usual way of package distribution. Feel free to pick some of the changes and close the PR.

```sh
npm run watch
npm run check-snapshots-new # in a separate terminal window
```

### CODE REVIEW

File: packages/pyright-scip/package.json [hunk 1]
```diff
@@ -7,8 +7,8 @@
     "build": "webpack --mode production --progress",
     "clean": "shx rm -rf ./dist ./out README.md LICENSE.txt",
     "prepack": "npm run clean && shx cp ../../README.md . && shx cp ../../LICENSE.txt . && npm run build",
-    "check-snapshots-new": "uvu ./test -r ts-node/register --ignore dist --check",
-    "update-snapshots-new": "uvu ./test -r ts-node/register --ignore dist",
+    "check-snapshots-new": "uvu ./test -r ts-node-transpile-only/register --check",
+    "update-snapshots-new": "uvu ./test -r ts-node-transpile-only/register --ignore dist",
```
Comment (valerybugakov) [inline packages/pyright-scip/package.json:11]: Do not spend time on type-checking while running tests

File: packages/pyright-scip/src/mainImpl.ts [hunk 2]
```diff
@@ -174,5 +174,3 @@ export function main(argv: string[]): void {
     });
     command.parse(argv);
 }
-
-main(process.argv);
```
Comment (valerybugakov) [inline packages/pyright-scip/src/mainImpl.ts:178]: This threw an error before I could call `main` in tests, so I deleted it temporarily. Do we need a separate entry-point for tests?

File: packages/pyright-scip/webpack.config.js [hunk 3]
```diff
@@ -26,6 +26,9 @@ module.exports = (_, { mode }) => {
         output: {
             filename: '[name].js',
             path: outPath,
+            library: {
+                type: 'commonjs2',
```
Comment (valerybugakov) [inline packages/pyright-scip/webpack.config.js:30]: Required to expose `exports` from the artifact.

File: packages/pyright-scip/package.json (hunk 1 above)
Comment (varungandhi-src) [reply packages/pyright-scip/package.json:11]: Hmm, does this require updating `package.json`? I can't tell which package this comes from... maybe it's being picked up from global state on your machine?

Comment (varungandhi-src) [reply packages/pyright-scip/package.json:11]: If this is not strictly necessary, I'm fine with sticking to `ts-node`

File: packages/pyright-scip/src/mainImpl.ts (hunk 2 above)
Comment (varungandhi-src) [reply packages/pyright-scip/src/mainImpl.ts:178]: This `main` call was incorrect, it should've been removed earlier, as there is already a call to `main` in `main.ts` which I thought would be the entry-point for the indexer itself, whereas `test-main.ts` would be the entry-point for the test code.

Comment (varungandhi-src) [reply packages/pyright-scip/src/mainImpl.ts:178]: I'm confused a little bit here... This patch removes both calls to `main`. That way, running `node ./index.js --help` doesn't print the help for the indexer itself...

Normally, in C++, I'd end up with two different binaries (one for releases, and one for tests) with two different `main` functions. The test binary would call the `mainImpl` from a shared library which contains 99% of the code.

Can we do something similar here/would that be unidiomatic?

**Pull request merged**